app/src/main/java/com/android/unio/model/association/AssociationViewModel.kt

```diff
@@ -94,17 +99,47 @@ constructor(
    * set to an empty list.
    */
   fun getAssociations() {
+    _refreshState.value = true
     associationRepository.getAssociations(
         onSuccess = { fetchedAssociations ->
           _associations.value = fetchedAssociations
           _associationsByCategory.value = fetchedAssociations.groupBy { it.category }
+          _refreshState.value = false
         },
         onFailure = { exception ->
           _associations.value = emptyList()
+          _associationsByCategory.value = emptyMap()
+          _refreshState.value = false
           Log.e("ExploreViewModel", "Failed to fetch associations", exception)
         })
   }
 
+  fun refreshAssociation() {
+    if (_selectedAssociation.value == null) {
+      return
+    }
+
+    _refreshState.value = true
+    associationRepository.getAssociationWithId(
+        _selectedAssociation.value!!.uid,
+        onSuccess = { fetchedAssociation ->
+          _selectedAssociation.value = fetchedAssociation
+          _selectedAssociation.value?.events?.requestAll()
+          _selectedAssociation.value?.members?.forEach { fetchUserFromMember(it) }
+
+          _associations.value =
+              _associations.value.map {
+                if (it.uid == fetchedAssociation.uid) fetchedAssociation else it
+              }
+          _associationsByCategory.value = _associations.value.groupBy { it.category }
+          _refreshState.value = false
+        },
+        onFailure = { exception ->
+          Log.e("AssociationViewModel", "Failed to fetch association", exception)
```

Comment:
I believe that a toast would be a good implementation in the onFailure to show the user what is going on